src/core/thread/topology.hpp

```diff
@@ -582,6 +607,9 @@ class Neighbor : public InstanceLocatorInit
         {
             uint32_t mLinkFrameCounter; ///< The Link Frame Counter
             uint32_t mMleFrameCounter;  ///< The MLE Frame Counter
+#if OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2
+            uint32_t mLinkAckFrameCounter; ///< The Link Ack Frame Counter
```

Comment:
Maybe in a future PR, we could save some memory here by not storing the full 32 bits. For example, I expect the difference between `mLinkFrameCounter` and `mLinkAckFrameCounter` to be small.

Reply:
Maybe we can use uint8_t to save the offset between two counter. However, in current structure, alignment padding might result in similar memory usage as uint32_t?

Reply:
I think it is fine to leave for now. In future, we may want the additional bytes for other features.